src/Schema/Directives/DropArgsDirective.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Nuwave\Lighthouse\Schema\Directives;
+
+use Closure;
+use GraphQL\Type\Definition\ResolveInfo;
+use Nuwave\Lighthouse\Execution\Arguments\ArgumentSet;
+use Nuwave\Lighthouse\Schema\Values\FieldValue;
+use Nuwave\Lighthouse\Support\Contracts\Directive;
+use Nuwave\Lighthouse\Support\Contracts\FieldMiddleware;
+use Nuwave\Lighthouse\Support\Contracts\GraphQLContext;
+use Nuwave\Lighthouse\Support\Utils;
+
+class DropArgsDirective extends BaseDirective implements FieldMiddleware
+{
+    public static function definition(): string
+    {
+        return /** @lang GraphQL */ <<<'GRAPHQL'
+"""
+Apply the @drop directives on the incoming arguments.
+"""
+directive @dropArgs on FIELD_DEFINITION
+GRAPHQL;
+    }
+
+    /**
+     * @param FieldValue $fieldValue
+     * @param Closure $next
+     * @return mixed
+     */
+    public function handleField(FieldValue $fieldValue, Closure $next)
+    {
+        $resolver = $fieldValue->getResolver();
+
+        return $next(
+            $fieldValue->setResolver(
+                function ($root, array $args, GraphQLContext $context, ResolveInfo $resolveInfo) use ($resolver) {
+                    $argumentSet = $resolveInfo->argumentSet;
+                    $this->drop($argumentSet);
+
+                    return $resolver(
+                        $root,
+                        $argumentSet->toArray(),
+                        $context,
+                        $resolveInfo
+                    );
+                }
+            )
+        );
+    }
+
+    /**
+     * @param ArgumentSet $old
+     * @return void
+     */
+    protected function drop(ArgumentSet &$old): void
```

Comment:
```suggestion
    protected function drop(ArgumentSet &$argumentSet): void
```

`$old` would make sense if we were constructing `$new`, but we mutate instead. You can change this accordingly in the rename directive